oauth2-server/src/main/java/com/clouway/oauth2/IssueNewTokenActivity.java

```diff
@@ -64,27 +72,27 @@ public Response execute(Client client, Request request, DateTime instant) {
     String certKey = randomKey(certificates);
     Pem.Block key = certificates.get(certKey);
 
-
     TokenResponse response = tokens.issueToken(GrantType.AUTHORIZATION_CODE, client, identity, authorization.scopes, instant);
     if (!response.isSuccessful()) {
       return OAuthError.invalidRequest("Token cannot be issued.");
     }
 
     BearerToken accessToken = response.accessToken;
-    String host = request.header("Host");
 
-    String idToken = newIdToken().header(new Header("RS256"))
-            .claims(newClaimSet()
-                    .iss(host)
-                    .aud(client.id)
-                    .sub(identity.id())
-                    .iat(instant.timestamp())
-                    .exp(accessToken.expirationTimestamp())
-                    .certId(certKey)
-                    .customClaims(identity.claims())
-                    .build())
-            .signWith(key)
-            .build();
+    String host = request.header("Host");
+    DefaultJwtBuilder idTokenBuilder = new DefaultJwtBuilder();
+    Gson gson = new Gson();
+
+    String idToken = idTokenBuilder
+            .setHeaderParam("cid", certKey)//CertificateId - the ID of the certificate that the token was signed with.
+            .setIssuer(host)
+            .setAudience(client.id)
+            .setSubject(identity.id())
+            .setIssuedAt(instant.asDate())
+            .setExpiration(new Date(accessToken.expirationTimestamp()))
+            .claim("identity", gson.toJson(identity))
```

Comment:
This will make it different from the UserInfo and will enforce clients to use different parses for each. 

This is how UserInfo claims are build:
```java
    o.addProperty("id", identity.id());
    o.addProperty("name", identity.name());
    o.addProperty("email", identity.email());
    o.addProperty("given_name", identity.givenName());
    o.addProperty("family_name", identity.familyName());

    Map<String, Object> claims = identity.claims();

    for (String key : claims.keySet()) {
      Object value = claims.get(key);

      if (value instanceof String) {
        o.addProperty(key, (String) value);
      }
      if (value instanceof Number) {
        o.addProperty(key, (Number) value);
      }

      if (value instanceof Boolean) {
        o.addProperty(key, (Boolean) value);
      }
    }
```